web-registry/src/hooks/useMsgClient.ts

```diff
@@ -35,26 +45,54 @@ export default function useMsgClient(
         ],
         gas: '200000',
       };
+
       try {
         const txBytes = await api.msgClient.sign(
           wallet.address,
           msgs,
           fee || defaultFee,
           memo || '',
         );
+
+        return txBytes;
+      } catch (err) {
+        setError(err);
+        return;
+      }
+    },
+    [api?.msgClient, wallet?.address],
+  );
+
+  const broadcast = useCallback(
+    async (txBytes: Uint8Array) => {
+      if (!api?.msgClient || !txBytes) return;
+      try {
+        handleTxQueued();
+        const _deliverTxResponse = await api.msgClient.broadcast(txBytes);
+        setDeliverTxResponse(_deliverTxResponse);
+        handleTxDelivered();
+      } catch (err) {
+        setError(err);
+      }
+    },
+    [api?.msgClient, handleTxQueued, handleTxDelivered],
+  );
+
+  const signAndBroadcast = useCallback(
+    async (message: MessageData, onBroadcast?: () => void) => {
+      try {
+        const txBytes = await sign(message);
         if (txBytes) {
-          handleTxQueued();
-          const _deliverTxResponse = await api.msgClient.broadcast(txBytes);
-          setDeliverTxResponse(_deliverTxResponse);
-          handleTxDelivered();
+          if (onBroadcast) onBroadcast();
```

Comment:
@blushi added this, to give us optional modal-closing after Keplr signing is confirmed.